compiler-rt/lib/memprof/memprof_mapping.h

```diff
@@ -104,8 +130,15 @@ inline bool AddrIsAlignedByGranularity(uptr a) {
 inline void RecordAccess(uptr a) {
   // If we use a different shadow size then the type below needs adjustment.
   CHECK_EQ(SHADOW_ENTRY_SIZE, 8);
-  u64 *shadow_address = (u64 *)MEM_TO_SHADOW(a);
-  (*shadow_address)++;
+  if (flags()->histogram) {
```

Comment:
I'm slightly concerned about the runtime overhead of the extra compare and branch for every memory access, but it probably isn't significant on top of the function call. I believe we still use the non-callback instrumentation by default, so this shouldn't affect default behavior iiuc (is that correct)? An alternative is to control this from the compiler, by having a separate version of RecordAccess (and its caller __memprof_* functions that are called by the instrumented code), but imo that isn't worth it if the default is unaffected.

Reply:
Solved this having two different functions, RecordAddress and RecordAddressWithHistogram, which get used based on compiler flag, not runtime flag.